src/library/scala/jdk/DoubleAccumulator.scala

```diff
@@ -428,7 +428,7 @@ private[jdk] class DoubleAccumulatorStepper(private val acc: DoubleAccumulator)
       ans
     }
 
-  override def spliterator: Spliterator.OfDouble = new DoubleStepper.DoubleStepperSpliterator(this) {
+  override def spliterator[B >: Double]: Spliterator.OfDouble = new DoubleStepper.DoubleStepperSpliterator(this) {
```

Comment:
Does this work when viewed from Java?  Does it need to work?

Reply:
It seems to work fine from Java. For a `DoubleStepper`, the return type is `OfDouble`, so things are ok.

If you have a `AnyStepper<X>` in Java, the `spliterator` method has signature `<B> Spliterator<B> spliterator()`. If there's an expected type Java's inference handles it fine, but you can write `myStringStepper.<Integer>spliterartor()` and get a CCE later on.

Reply:
Why isn't the Java signature `<B super java.lang.Double> Spliterator<B>` in the first place?